test/lib/waitUntilCalled.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * A function that wraps a sinon stubbed function and returns a Promise
+ * when this stub was called.
+ *
+ * The stub that has been passed in will be setup to call the wrapped function
+ * directly, then trigger the returned Promise to resolve.
+ *
+ * @param {Object} stub - A sinon stub of a function
```

Comment:
I think this could be helpful here:
```suggestion
 * @param {import('sinon').stub} stub - A sinon stub of a function
```

Reply:
Good point! This does seem good. I have seen this sort of syntax used before, but I don't really understand it. How does this work - is this part of some standard, or a TypeScript thing, or?

Reply:
I think it's a TypeScript thing, @brad-decker introduced me to it. VS Code understands it, as do other common editors. It doesn't appear to be supported by JSDoc just yet, although I don't think that should stop us: https://github.com/jsdoc/jsdoc/issues/1645

Reply:
Addressed here: https://github.com/MetaMask/metamask-extension/pull/9755/commits/9fcd15ab0f8957f01ccea8fd370b4df184c0314f